src/masks/InteractionMasks.tsx

```diff
@@ -140,94 +122,104 @@ export default function InteractionMasks<R>({
-  }
 
-  function onKeyDown(e: React.KeyboardEvent<HTMLDivElement>): void {
-    if (isCtrlKeyHeldDown(e)) {
-      onPressKeyWithCtrl(e);
-    } else if (e.keyCode === KeyCodes.Escape) {
-      onPressEscape();
-    } else if (e.keyCode === KeyCodes.Tab) {
-      onPressTab(e);
-    } else if (isKeyboardNavigationEvent(e)) {
-      changeCellFromEvent(e);
-    } else if (isKeyPrintable(e.keyCode) || [KeyCodes.Backspace, KeyCodes.Delete, KeyCodes.Enter].includes(e.keyCode)) {
-      openEditor(e);
-    }
+    return getNextSelectedCellPosition({
+      columns,
+      rowsCount: rows.length,
+      cellNavigationMode: mode,
+      nextPosition
+    });
   }
 
-  function openEditor(event: React.KeyboardEvent<HTMLDivElement>): void {
-    if (!isEditorEnabled && isCellEditable(selectedPosition)) {
-      setFirstEditorKeyPress(event.key);
-      setIsEditorEnabled(true);
+  function onKeyDown(event: React.KeyboardEvent<HTMLDivElement>): void {
+    const column = columns[selectedPosition.idx];
+    const row = rows[selectedPosition.rowIdx];
+    const isActivatedByUser = (column.unsafe_onCellInput ?? legacyCellInput)(event, row) === true;
+
+    const { key } = event;
+    if (enableCellCopyPaste && isCtrlKeyHeldDown(event)) {
+      // event.key may be uppercase `C` or `V`
+      const lowerCaseKey = event.key.toLowerCase();
+      if (lowerCaseKey === 'c') return handleCopy();
+      if (lowerCaseKey === 'v') return handlePaste();
     }
-  }
-
-  function closeEditor(): void {
-    setIsEditorEnabled(false);
-    setFirstEditorKeyPress(null);
-  }
 
-  function onPressKeyWithCtrl({ keyCode }: React.KeyboardEvent<HTMLDivElement>): void {
-    if (!enableCellCopyPaste) return;
-    if (keyCode === KeyCodes.c) {
-      handleCopy();
-    } else if (keyCode === KeyCodes.v) {
-      handlePaste();
+    const canOpenEditor = selectedPosition.status === 'SELECT' && isCellEditable(selectedPosition);
+
+    switch (key) {
+      case 'Enter':
+        if (canOpenEditor) {
+          setSelectedPosition(({ idx, rowIdx }) => ({ idx, rowIdx, status: 'EDIT', key: 'Enter' }));
+        } else if (selectedPosition.status === 'EDIT') {
+          setSelectedPosition(({ idx, rowIdx }) => ({ idx, rowIdx, status: 'SELECT' }));
+        }
+        break;
+      case 'Escape':
+        closeEditor();
+        setCopiedPosition(null);
+        break;
+      case 'Tab':
+        onPressTab(event);
+        break;
+      case 'ArrowUp':
+      case 'ArrowDown':
+      case 'ArrowLeft':
+      case 'ArrowRight':
+        event.preventDefault();
+        selectCell(getNextPosition(key));
+        break;
+      default:
+        if (canOpenEditor && isActivatedByUser) {
+          setSelectedPosition(({ idx, rowIdx }) => ({ idx, rowIdx, status: 'EDIT', key }));
+        }
+        break;
     }
   }
 
   function onPressTab(e: React.KeyboardEvent<HTMLDivElement>): void {
-    // When there are no rows in the grid, we need to allow the browser to handle tab presses
```

Comment:
I do not think this is possible. We show selection mask if the cell is within row/column bound and if there are no rows then `selectionMask` is not rendered. We set the `tabIndex` on the selection mask so `onKeyDown` will never be trigged